File: UIDemo/VideoDEMO/src/domain/models/video-state.ts
/**
 * Represents the MassTransit saga state for a video processing pipeline.
 * Maps to the backend VideoState entity in Infrastructure/Sagas/VideoState.cs
 */
export interface VideoState {
  correlationId: string;
  tenantId: string;
  currentState: string;
  mediaConverterJobId?: string;
  key?: string;
  sourceWidth?: number;
  sourceHeight?: number;
  duration?: string;
  s3OutputPrefix?: string;
  finalUrl?: string;
  encryptionMethod?: string;
  customThumbnailKey?: string;
  targetResourceArn?: string;
  isFailed: boolean;
  version: number;
}

/**
 * Real-time progress update received via SignalR VideoProgressHub.
 * Maps to VideoProgressNotification in Application/Services/IVideoProgressNotifier.cs
 */
export interface VideoProgressUpdate {
  videoId: string;
  tenantId: string;
  stage: 'uploading' | 'analyzing' | 'preparing' | 'transcoding' | 'publishing' | string;
  status: 'IN_PROGRESS' | 'COMPLETE' | 'FAILED' | string;
  percentage: number | null;
  metadata: string | null;
}

File: src/alphazero-api/Modules/VideoUploading/Infrastructure/Sagas/VideoState.cs
using MassTransit;

namespace AlphaZero.Modules.VideoUploading.Infrastructure.Sagas;

public class VideoState : SagaStateMachineInstance
{
    public Guid CorrelationId { get; set; }
    public Guid TenantId { get; set; }
    public string CurrentState { get; set; } = null!;
    public string? MediaConverterJobId { get; set; }
    public string? Key { get; set; }
    public int? SourceWidth { get; set; }
    public int? SourceHeight { get; set; }
    public TimeSpan? Duration { get; set; }
    public string? S3OutputPrefix { get; set; }
    public string? FinalUrl { get; set; }
    public string? EncryptionMethod { get; set; }
    public string? CustomThumbnailKey { get; set; }
    public string? TargetResourceArn { get; set; }
    public bool IsFailed { get; set; } = false;
    public int Version { get; set; }
}
